src/Core/Asset/GeometryData.inl

```diff
@@ -188,47 +197,191 @@ inline bool GeometryData::isHexMesh() const {
+    auto& attrib = getAttrib<V>( name );
+    auto& data   = attrib.getDataWithLock();
+    internal::copyData( attribDataList, data );
+    attrib.unlock();
+}
+
+template <typename V>
+inline VectorArray<V>&
+GeometryData::getDataFromLayerBase( Geometry::GeometryIndexLayerBase& geomBase ) {
+    auto& v    = dynamic_cast<Geometry::GeometryIndexLayer<V>&>( geomBase );
+    auto& data = v.collection();
+    return data;
+}
+
+template <typename V>
+inline const VectorArray<V>&
+GeometryData::getDataFromLayerBase( const Geometry::GeometryIndexLayerBase& geomBase ) const {
+    const auto& v    = dynamic_cast<const Geometry::GeometryIndexLayer<V>&>( geomBase );
+    const auto& data = v.collection();
+    return data;
+}
+
+template <typename L>
+inline Geometry::GeometryIndexLayerBase&
+GeometryData::getLayerBaseWithLock( const bool& firstOccurrence, const std::string& name ) {
+    auto& geomBase =
+        ( firstOccurrence )
+            ? m_multiIndexedGeometry.getFirstLayerOccurrenceWithLock( L::staticSemanticName ).second
+            : m_multiIndexedGeometry.getLayerWithLock( { L::staticSemanticName }, name );
+    return geomBase;
+}
+
+template <typename L>
+inline const Geometry::GeometryIndexLayerBase&
+GeometryData::getLayerBase( const bool& firstOccurrence, const std::string& name ) const {
+    const auto& geomBase =
+        ( firstOccurrence )
+            ? m_multiIndexedGeometry.getFirstLayerOccurrence( L::staticSemanticName ).second
+            : m_multiIndexedGeometry.getLayer( { L::staticSemanticName }, name );
+    return geomBase;
+}
+
+template <typename V, typename L>
+inline VectorArray<V>& GeometryData::getIndexedDataWithLock( const bool& firstOccurrence,
+                                                             const std::string& name ) {
+    if ( m_multiIndexedGeometry.containsLayer( { L::staticSemanticName }, name ) ) {
+        auto& geomBaseFound = getLayerBaseWithLock<L>( firstOccurrence, name );
+        return getDataFromLayerBase<V>( geomBaseFound );
+    }
+    else {
+        auto pil = std::make_unique<L>();
+        auto& geomBaseCreated =
+            m_multiIndexedGeometry.addLayer( std::move( pil ), true, name ).second;
+        return getDataFromLayerBase<V>( geomBaseCreated );
+    }
+}
+
+template <typename V>
+inline VectorArray<V>& GeometryData::findIndexDataWithLock( const std::string& name,
+                                                            const bool& firstOccurrence ) {
+    if ( std::is_same<V, Vector2ui>::value ) {
+        return getIndexedDataWithLock<V, Geometry::LineIndexLayer>( firstOccurrence, name );
+    }
+    /* else if (std::is_same<V, Vector3ui>::value){
+        return getIndexedDataWithLock<V, Geometry::TriangleIndexLayer>( firstOccurrence, name );
+    }
+    else if (std::is_same<V, Vector4ui>::value){
+        return getIndexedDataWithLock<V, Geometry::QuadIndexLayer>( firstOccurrence, name );
+    } */
```

Comment:
remove uneeded comment ?

Reply:
In fact this is useless but it show what kind of modifications will be able in my next pr. So, as you wish.

Reply:
Ok i will do that.